bpf/bpf_lxc.c

```diff
@@ -1289,7 +1290,14 @@ ipv6_policy(struct __ctx_buff *ctx, int ifindex, __u32 src_label,
 	}
 
 #ifdef ENABLE_NODEPORT
-	if (ret == CT_NEW || ret == CT_REOPENED) {
+	if (tuple.nexthdr == IPPROTO_TCP) {
+		if (ctx_load_bytes(ctx, l4_off + 12, &tcp_flags, 2) < 0)
+			return DROP_CT_INVALID_HDR;
+	}
+	if (ret == CT_NEW || ret == CT_REOPENED ||
+	    /* See comment in ipv4_policy() */
+	    (ret == CT_ESTABLISHED && (tuple.nexthdr != IPPROTO_TCP ||
```

Comment:
We don't need `tuple.nexthdr != IPPROTO_TCP` I think. `(tcp_flags.value & TCP_FLAG_SYN) && !(tcp_flags.value & TCP_FLAG_ACK))` will never become true when `tuple.nexthdr != IPPROTO_TCP`. We already check it here.

https://github.com/cilium/cilium/pull/20006/files#diff-1e6ee3d7f22bd39ab87f7d8d17b8f1c909de590f8a2f694e32a6508fad98103bR1293

Reply:
I think we need to override a ct entry as DSR when

- proto = TCP and SYN and !ACK
- proto != TCP, e.g. UDP

Without `tuple.nexthdr != IPPROTO_TCP`, we can't do it with UDP packets.